.github/workflows/application-signals-python-e2e-ec2-test.yml

```diff
@@ -0,0 +1,253 @@
+## Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+## SPDX-License-Identifier: Apache-2.0
+
+# This is a reusable workflow for running the Python E2E Canary test for Application Signals.
+# It is meant to be called from another workflow.
+# Read more about reusable workflows: https://docs.github.com/en/actions/using-workflows/reusing-workflows#overview
+name: Application Signals Enablement E2E Testing - Python EC2 Use Case
+on:
+  workflow_call:
+    inputs:
+      aws-region:
+        required: true
+        type: string
+      staging_wheel_name:
+        required: false
+        default: 'aws-opentelemetry-distro'
+        type: string
+      caller-workflow-name:
+        required: true
+        type: string
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  SAMPLE_APP_ZIP: s3://${{ secrets.APP_SIGNALS_E2E_EC2_JAR }}-prod-${{ inputs.aws-region }}/python-sample-app.zip
+  METRIC_NAMESPACE: AppSignals
+  LOG_GROUP_NAME: /aws/appsignals/generic
+  ADOT_WHEEL_NAME: ${{ inputs.staging_wheel_name }}
+  TEST_RESOURCES_FOLDER: ${GITHUB_WORKSPACE}
+
+jobs:
+  python-e2e-ec2-test:
+    runs-on: ubuntu-latest
+    container:
+      image: public.ecr.aws/h6o3z5z9/aws-application-signals-test-framework-workflow-container:latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set CW Agent RPM environment variable
+        run: echo GET_CW_AGENT_RPM_COMMAND="wget -O cw-agent.rpm https://amazoncloudwatch-agent-${{ inputs.aws-region }}.s3.${{ inputs.aws-region }}.amazonaws.com/amazon_linux/amd64/latest/amazon-cloudwatch-agent.rpm" >> $GITHUB_ENV
+
+      - name: Generate testing id
+        run: echo TESTING_ID="${{ github.job }}-${{ env.AWS_DEFAULT_REGION }}-${{ github.run_id }}-${{ github.run_number }}-${{ github.run_attempt }}" >> $GITHUB_ENV
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.E2E_SECRET_TEST_ROLE_ARN }}
+          aws-region: us-east-1
+
+      - name: Retrieve account
+        uses: aws-actions/aws-secretsmanager-get-secrets@v1
+        with:
+          secret-ids:
+            ACCOUNT_ID, region-account/${{ inputs.aws-region }}
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: arn:aws:iam::${{ env.ACCOUNT_ID }}:role/${{ secrets.E2E_TEST_ROLE_ARN }}
+          aws-region: ${{ inputs.aws-region }}
+
+      - uses: actions/download-artifact@v3
+        if: inputs.caller-workflow-name == 'main-build'
+        with:
+          name: ${{ inputs.staging_wheel_name }}
```

Comment:
For Java, the release testing code (workflow for building a new adot artifact and testing it) is located in the OTEL repository (https://github.com/aws-observability/aws-otel-java-instrumentation/blob/main/.github/workflows/appsignals-e2e-eks-test.yml) Is there a reason why the 'main-build workflow' is here for Python? Also, where is the workflow located for building and uploading the new adot artifact?

Reply:
We want to re-use the same workflow file for main_build in adot python repo to save maintenance effort, so I'm combing canary test and main_build E2E test in one file. In this case, there will be no [application-signals-python-e2e-ec2-test.yml](https://github.com/aws-observability/aws-otel-python-instrumentation/blob/main/.github/workflows/application-signals-python-e2e-ec2-test.yml) file exist. Same applies for EKS test. 

For building and uploading new artifact, they are both done in [main_build.yml](https://github.com/aws-observability/aws-otel-python-instrumentation/blob/main/.github/workflows/main_build.yml) work flow in adot python repo. The wheel file is upload to [S3 bucket](https://github.com/aws-observability/aws-otel-python-instrumentation/blob/1f13ce8202fd72fe9edb421ec2b4948709b26195/.github/workflows/main_build.yml#L75C11-L75C114), and the image is pushed to [ECR repository](https://github.com/aws-observability/aws-otel-python-instrumentation/blob/1f13ce8202fd72fe9edb421ec2b4948709b26195/.github/workflows/main_build.yml#L50).